delfin/drivers/netapp/dataontap/netapp_handler.py

```diff
@@ -465,3 +480,414 @@ def get_filesystems(self, storage_id):
                 }
                 fs_list.append(fs_model)
         return fs_list
+
+    def list_controllers(self, storage_id):
+        try:
+            controller_list = []
+            controller_info = self.ssh_pool.do_exec(
+                constant.CONTROLLER_SHOW_DETAIL_COMMAND)
+            controller_array = controller_info.split(
+                constant.CONTROLLER_SPLIT_STR)
+            controller_map = {}
+            for controller_str in controller_array[1:]:
+                Tools.split_value_map(
+                    controller_str, controller_map, split=':')
+                if controller_map is not None or controller_map != {}:
+                    status = constants.ControllerStatus.NORMAL \
+                        if controller_map['Health'] == 'true' \
+                        else constants.ControllerStatus.OFFLINE
+                    controller_model = {
+                        'name': controller_map['e'],
```

Comment:
what is value for  key 'e' ? do you have an exmplae of command output and parsed  map ?

Reply:
The sample is：
                                              Node: cl-01
                                             Owner:
                                          Location:
                                             Model: SIMBOX
                                     Serial Number: 4082368-50-7
                                         Asset Tag: -
                                            Uptime: 3 days 21:20
                                   NVRAM System ID: 4082368507
                                         System ID: 4082368507
                                            Vendor: NetApp
                                            Health: true
                                       Eligibility: true
                           Differentiated Services: false
                               All-Flash Optimized: false

Here the first Nod is removed as a split string, so 'e' is used as the key

Reply:
Mappings have been added for easy reading